File: leetcode/Remove_Duplicates_from_Sorted_Array_II.java
import java.util.*;

public class Solution {
    public int removeDuplicates(int[] A) {
        if (A.length == 0) return 0;
        int cur = 0;
        int ct = 1;
        for (int i = 1; i < A.length; ++i) {
            if (A[cur] != A[i]) {
                ++cur;
                A[cur] = A[i];
                ct = 1;
            } else if (A[cur] == A[i] && ct == 1) {
                ++cur;
                A[cur] = A[i];
                ct++;
            }
        }
        return cur + 1;
    }
}

File: leetcode/Remove_Duplicates_from_Sorted_Array_II.cpp
class Solution {
public:
    int removeDuplicates(int A[], int n) {
        int new_n = 0;
        int current_val = numeric_limits<int>::min();
        int count = 0;
        for (int i = 0; i < n; ++i) {
            if (A[i] != current_val) {
                A[new_n++] = current_val = A[i];
                count = 1;
            } else {
                ++count;
                if (count == 2)
                    A[new_n++] = A[i];
            }
        }
        return new_n;
    }
};
